apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -676,15 +676,116 @@ function openLinksInParentFrame( calypsoPort ) {
 		} );
 	} );
 
-	if ( calypsoifyGutenberg.manageReusableBlocksUrl ) {
-		const manageReusableBlocksLinkSelectors = [
-			'.block-editor-inserter__manage-reusable-blocks', // Link in the Blocks Inserter
-			'a.components-menu-item__button[href*="post_type=wp_block"]', // Link in the More Menu
-		].join( ',' );
-		$( '#editor' ).on( 'click', manageReusableBlocksLinkSelectors, ( e ) => {
-			e.preventDefault();
-			window.open( calypsoifyGutenberg.manageReusableBlocksUrl, '_top' );
+	const { createNewPostUrl, manageReusableBlocksUrl } = calypsoifyGutenberg;
+	if ( ! createNewPostUrl && ! manageReusableBlocksUrl ) {
+		return;
+	}
+
+	await isEditorReadyWithBlocks();
+
+	// Create a new post link in block settings sidebar for Query block
+	const createNewPostLinkObserver = new window.MutationObserver( () => {
+		const hyperlink = document.querySelector( '.wp-block-query__create-new-link a' );
+		if ( hyperlink ) {
+			hyperlink.href = createNewPostUrl;
+			hyperlink.target = '_top';
+		}
+	} );
+
+	// Manage reusable blocks link in the global block inserter's Reusable tab
+	// Post editor only
+	const inserterManageReusableBlocksObserver = new window.MutationObserver( ( mutations ) => {
+		const node = mutations[ 0 ].target;
+		if ( node.ariaSelected === 'true' ) {
+			const hyperlink = document.querySelector( 'a.block-editor-inserter__manage-reusable-blocks' );
+			if ( hyperlink ) {
+				hyperlink.href = manageReusableBlocksUrl;
+				hyperlink.target = '_top';
+			}
+		}
+	} );
+
+	const shouldReplaceCreateNewPostLinksFor = ( node ) =>
+		createNewPostUrl &&
+		( node.classList.contains( 'interface-interface-skeleton__sidebar' ) || // Site editor
+			node.classList.contains( 'edit-post-sidebar' ) ); // Post editor
+
+	const shouldReplaceManageReusableBlockLinksFor = ( node ) =>
+		manageReusableBlocksUrl &&
+		node.classList.contains( 'interface-interface-skeleton__secondary-sidebar' );
+
+	const sidebarsObserver = new window.MutationObserver( ( mutations ) => {
+		for ( const record of mutations ) {
+			for ( const node of record.addedNodes ) {
+				if ( shouldReplaceCreateNewPostLinksFor( node ) ) {
```

Comment:
What do you think about disconnecting the `createNewPostLinkObserver` when `shouldReplaceCreateNewPostLinksFor` is false? (and similar for the other observers) It seems like it would be a good idea to clean these up when the relevant elements are removed from the DOM.

Reply:
Oh, now I see you're already doing that down on line 740.